### PR TITLE
Migrate crud.md to a compiled java file

### PR DESCRIPTION
**Pull request by tomblench**
See #405 

To compile, run `gradle cloudant-sync-datastore-core:compileCrudSamples`

### CODE REVIEW

File: doc/CrudSamples.java [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+        File path = new File("/tmp");
+        // on Android, we could do something like:
+        // File path = getApplicationContext().getDir("document_stores");
+
+        // Once you've got a path, it's straightforward to create DocumentStores:
+
+        DocumentStore ds = DocumentStore.getInstance(new File(path, "my_document_store"));
+        DocumentStore ds2 = DocumentStore.getInstance(new File(path, "other_document_store"));
+
+        // Static getInstance() methods on DocumentStore will create and initial a DocumentStore
```
Comment (ricellis) [inline doc/CrudSamples.java]: initialize?

File: doc/CrudSamples.java [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+        // instance if it doesn't already exist, or retrieve an existing one.
+
+        // To close a DocumentStore (which closes the underlying database files and release native
```
Comment (ricellis) [inline doc/CrudSamples.java]: releases

File: doc/CrudSamples.java [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+        // resources):
+        ds.close();
```
Comment (ricellis) [inline doc/CrudSamples.java]: Maybe close and delete should be at the end for a more lifecycle oriented flow of reading the examples?

File: doc/CrudSamples.java [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+
+        // To delete a DocumentStore (which closes it first):
+        ds.delete();
+
+        // Document CRUD APIs
+
+        // Once you have a DocumentStore instance, you can use it to create, update and
```
Comment (ricellis) [inline doc/CrudSamples.java]: what about read? admittedly if you've just created the store there won't be anything to read yet.

File: doc/CrudSamples.java [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+        // delete documents
+
+        // Create
+
+        // Documents are represented as a set of revisions. To create a document, you
+        // set up the initial revision of the document and save that to the datastore.
+
+        // Create a document revision object, set its body, ID and attachments
+        // and then call createDocumentFromRevision(DocumentRevision) to add it to the datastore:
```
Comment (ricellis) [inline doc/CrudSamples.java]: needs the new create method name

File: doc/CrudSamples.java [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+        json.put("completed", false);
+        json.put("type", "com.cloudant.sync.example.task");
+        rev.setBody(DocumentBodyFactory.create(json));
+
+        // Now call create(). Note that we call all of the CRUD methods through the Database
+        // instance which is obtained via the database() getter.
+        DocumentRevision revision = ds.database().create(rev);
+
+        // The only mandatory property to set before calling
+        // createDocumentFromRevision(DocumentRevision) is the body. An ID will be generated
```
Comment (ricellis) [inline doc/CrudSamples.java]: create

File: doc/CrudSamples.java [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+        retrieved.setBody(DocumentBodyFactory.create(json));
+
+        // Note that "updated" is the new DocumentRevision with a new revision ID
+        DocumentRevision updated = ds.database().update(retrieved);
+
+        // Delete
+
+        // To delete a document, you need the current revision:
+
+        ds.database().delete(retrieved);
```
Comment (ricellis) [inline doc/CrudSamples.java]: don't we also have a delete for specific revisions?

File: doc/CrudSamples.java [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+
+        // Indexing and Query
+
+        // You don't need to know the ID of the document to retrieve it. The DocumentStore
+        // provides ways to index and search the fields of your JSON documents.
+        // For more, see https://github.com/cloudant/sync-android/blob/master/doc/query.md.
+
+        // Conflicts
```
Comment (ricellis) [inline doc/CrudSamples.java]: This needs some content or a link or deleting?

File: doc/CrudSamples.java [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+
+        DocumentRevision saved = ds.database().create(rev);
+
+        // When creating new attachments, use `UnsavedFileAttachment` for data you already have on
+        // disk. Use
+        // `UnsavedStreamAttachment` when you have data which comes from an InputStream or is
+        // already in
+        // memory.
+
+        att1 = new UnsavedFileAttachment(
```
Comment (ricellis) [inline doc/CrudSamples.java]: this duplicates the previous

File: doc/CrudSamples.java [hunk 10]
```diff
@@ -0,0 +1,388 @@
...
+        // To remove an attachment, remove it from the `attachments` map:
+
+        retrieved = ds.database().read("myDoc");
+        retrieved.getAttachments().remove("cute_cat.jpg");
+        updated = ds.database().update(retrieved);
+
+        // To remove all attachments, set the `attachments` property to an empty map
+        // or `null`:
+
+        updated.setAttachments(null);
```
Comment (ricellis) [inline doc/CrudSamples.java]: Given that the previous shows the `ds.database().update()` required to persist this change, this probably should as well.

File: doc/CrudSamples.java (hunk 7 above)
Comment (ricellis) [inline doc/CrudSamples.java]: Ah I see it is at the end :)

File: doc/CrudSamples.java [hunk 11]
```diff
@@ -180,7 +179,7 @@ public DocumentRevision resolve(String docId, List<? extends DocumentRevision>
 
         DocumentRevision saved = ds.database().create(rev);
 
-        // Above, we used UnsavedFileAttachment for data which was already on. Use
+        // Above, we used UnsavedFileAttachment for data which was already on. disk Use
```
Comment (brynh) [inline doc/CrudSamples.java]: `on. disk` -> `on disk.`

**Pull request merged**